torch/fx/experimental/proxy_tensor.py

```diff
@@ -284,7 +286,17 @@ def __repr__(self):
 
     @classmethod
     def __torch_dispatch__(cls, func_overload, types, args=(), kwargs=None):
-        return proxy_call(func_overload, args, kwargs)
+        # Verify that the proxy mode for all arguments is the same
+        proxy_mode = None
+        for arg in pytree.tree_flatten((args, kwargs))[0]:
+            if isinstance(arg, ProxyTensor):
+                if proxy_mode is None:
+                    proxy_mode = arg.proxy_mode
+                else:
+                    assert proxy_mode is arg.proxy_mode, "All arguments must be in the same proxy mode"
```

Comment:
Small nit that Proxy also checks that all the tracers are the same (equivalent of this). I'm not attached to either one but if we want to have the same error message as before, we can just break after we find the first proxy mode and assume that they're all the same and let Proxy deal with it if that's not the case

Reply:
Sure, I can change that.

Reply:
Actually, @samdow, didn't you write some utilities for doing this? Or am I misremembering

Reply:
Oh actually yeah--if we need to do something like this in the future, we just need to map to get all the proxy_modes and then call `all_modes_same_scope(modes)` (https://github.com/pytorch/pytorch/blob/master/torch/utils/_mode_utils.py#L124)